specs/networking/libp2p-standardization.md

```diff
@@ -0,0 +1,192 @@
+ETH 2.0 Networking Spec - Libp2p standard protocols
+===
+
+# Abstract
+
+Ethereum 2.0 clients plan to use the libp2p protocol networking stack for
+mainnet release. This document aims to standardize the libp2p client protocols,
+configuration and messaging formats.
+
+# Libp2p Protocols
+
+## Transport
+
+This section details the libp2p transport layer that underlies the
+[protocols](#protocols) that are listed in this document.
+
+Libp2p allows composition of multiple transports. Eth2.0 clients should support
+TCP/IP and optionally websockets. Websockets are useful for implementations
+running in the browser and therefore native clients would ideally support these implementations 
+by supporting websockets.
+
+An ideal libp2p transport would therefore be TCP/IP with a fallback to
+websockets.
+
+### Encryption
+
+Libp2p currently offers [Secio](https://github.com/libp2p/specs/pull/106) which
```

Comment:
libp2p now has a [spec for TLS 1.3](https://github.com/libp2p/specs/tree/master/tls), with a [reference implementation in Go](https://github.com/libp2p/go-libp2p-tls). It's currently undergoing an [experiment in IPFS](https://github.com/ipfs/go-ipfs/pull/6229), and once we deem it stable, we'll recommend TLS 1.3 over SecIO.

It's OK to start with SecIO for now, but we'll recommend and switch to TLS 1.3 in IPFS and Filecoin as a default once it's stable.

We're also have a security firm auditing TLS 1.3, amongst other libp2p core components.

Reply:
One aspect that worries us is that libp2p peers negotiate about which encryption method to use in cleartext during connection bootstrapping. Consequently we may be vulnerable to censorship via deep packet inspection. We aim to make our protocols indistinguishable from HTTP/3, hence the preference for TLS 1.3.

Reply:
Totally agree. I wasn't aware how far along TLS 1.3 was. I've currently laid out a few protocols/choices here, mainly as a template so others can discuss which is the best to proceed with. I agree TLS 1.3 is preferred over Secio. Will add this into the section. For initial multiclient testnets, perhaps we start with secio, then upgrade.